rfcs/0020-telemetry.md

```diff
@@ -0,0 +1,192 @@
+
+**Note**
+
+There is no reason you cannot combine the two methods.
+You could use the HEC for your Elixir applications and the UF for non-Elixir applications.
+
+### Splunk Universal Forwarder
+
+The Splunk Universal Forwarder (UF) is an agent that runs alongside an application, takes in metrics in one of many formats, and forwards those metrics to Splunk.
+The UF can be used with any application, regardless of language, which makes it the most flexible way to get metrics into Splunk.
+
+It runs in a [sidecar](https://www.oreilly.com/library/view/designing-distributed-systems/9781491983638/ch02.html) to any container that emits metrics.
+
+To use it, create a metrics index in Splunk and add the sidecar to the `container_definition_json` of your `aws-ecs-container-definition` module:
+```terraform
+container_definition_json = jsonencode([
+    module.your-container.json_map_object,
+    module.sidecar-container.json_map_object
+])
+```
+
+You use the StatsD telemetry reporter highlighted above to emit your metrics in the StatsD format, and the UF will automatically forward them.
+
+#### When you should choose this method
+
+- You maintain applications that aren't written in Elixir or want all of your applications to emit metrics in a standard format.
+
+Every language has a library that emits metrics in the StatsD format; they are straightforward to set up.
+You get a lot of flexibility with very little work by running the UF as a sidecar and using an intermediary standard like StatsD.
+
+#### When you shouldn't choose this method
+
+- You only need metrics from Elixir applications.
+- You are hesitant to change your infrastructure.
+- You are concerned about your ECS resource usage.
+
+The UF sidecar uses CPU and memory resources.
+Though the footprint is small, you might be worried that increasing your compute usage might affect the rest of your cluster.
+Resource usage is more pertinent if you run resource-lite instances, as the UF would take up a more significant percentage of CPU and memory.
+Note that you can always increase CPU and memory levels for your cluster.
+
+### Splunk HTTP Event Collector
+
+The Splunk HTTP Event Collector (HEC) allows us to submit metrics via HTTP.
+We maintain a library [TelemetryMetricsSplunk](https://github.com/anthonyshull/telemetry_metrics_splunk) that forwards telemetry metrics to Splunk via the HEC.
+
+To use it, create a metrics index in Splunk and add its HEC token to the library configuration.
+
+```elixir
+config :app, TelemetryMetricsSplunk,
+  index: "your-index"
+  token: System.get_env("SPLUNK_TOKEN")
+```
+
+#### When you should choose this method
+
+- You only need metrics from Elixir applications.
+
+#### When you shouldn't choose this method
+
+- You maintain non-Elixir applications and do not want to write a method of getting metrics into Splunk.
+
+There aren't any libraries in other languages that transmit metrics to Splunk via the HEC.
```

Comment:
question: if we're maintaining a library to do this for Elixir, why wouldn't we maintain one for other languages we use?